fdbserver/workloads/SpecialKeySpaceCorrectness.actor.cpp

```diff
@@ -1049,18 +1051,22 @@ struct SpecialKeySpaceCorrectnessWorkload : TestWorkload {
 					tx->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 					Optional<Value> res = wait(tx->get(coordinatorsKey));
 					ASSERT(res.present()); // Otherwise, database is in a bad state
-					ClusterConnectionString cs(res.get().toString());
-					ASSERT(cs.coordinators().size() == old_coordinators_processes.size() + 1);
+					state ClusterConnectionString csNew(res.get().toString());
+					ASSERT(csNew.coordinators().size() + csNew.hostnames.size() ==
```

Comment:
Won't this assertion fail if `csNew.hasUnresolvedHostnames` == false?

I think what you want is `ASSERT(cs.coordinators().size() == old_coordinators_processes.size() + 1);` after `wait(csNew.resolveHostnames());`

Reply:
They are equivalent. If `csNew.hasUnresolvedHostnames == false`, then `cs.coordinators().size()=0`. But I did change to the way you suggested.